File: examples/text_generation.py
"""
Example script demonstrating text generation with Pollinations.

This script shows various ways to use the text generation API.
"""

from pollinations import Pollinations


def main():
    # Create a client
    client = Pollinations()
    
    print("=" * 60)
    print("Text Generation Examples")
    print("=" * 60)
    
    # Example 1: Simple text generation
    print("\n1. Simple text generation:")
    print("-" * 60)
    response = client.generate_text("What is artificial intelligence?")
    print(response)
    
    # Example 2: With system message
    print("\n2. With system message:")
    print("-" * 60)
    response = client.generate_text(
        "Write a haiku about programming",
        system="You are a creative poetry assistant who writes beautiful haikus"
    )
    print(response)
    
    # Example 3: With temperature control
    print("\n3. Creative response (high temperature):")
    print("-" * 60)
    response = client.generate_text(
        "Tell me a short story about a robot",
        temperature=0.9
    )
    print(response)
    
    # Example 4: Deterministic response with seed
    print("\n4. Deterministic response (with seed):")
    print("-" * 60)
    response1 = client.generate_text("Count to 5", seed=42)
    print(f"First call: {response1}")
    response2 = client.generate_text("Count to 5", seed=42)
    print(f"Second call: {response2}")
    print(f"Same output: {response1 == response2}")
    
    # Example 5: JSON mode
    print("\n5. JSON mode:")
    print("-" * 60)
    response = client.generate_text(
        "Create a JSON object with name, age, and city fields for a fictional person",
        jsonMode=True
    )
    print(response)
    
    # Example 6: Using specific model
    print("\n6. Using specific model:")
    print("-" * 60)
    try:
        response = client.generate_text(
            "Explain quantum entanglement in simple terms",
            model="openai"
        )
        print(response)
    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    main()

File: examples/javascript/text_generation.js
/**
 * Example demonstrating basic text generation.
 * 
 * This script shows simple text generation examples.
 */

import { Pollinations } from '@gpt4free/g4f.dev';

async function main() {
  console.log('='.repeat(70));
  console.log('Text Generation Examples');
  console.log('='.repeat(70));
  
  // Initialize client
  const client = new Pollinations();
  
  // ========================================================================
  // Basic Text Generation
  // ========================================================================
  
  console.log('\n1. Simple Question:');
  console.log('-'.repeat(70));
  const response1 = await client.chat.completions.create({
    messages: [
      { role: 'user', content: 'What is the meaning of life?' }
    ]
  });
  console.log(response1.choices[0].message.content);
  
  console.log('\n2. With System Message:');
  console.log('-'.repeat(70));
  const response2 = await client.chat.completions.create({
    messages: [
      { role: 'system', content: 'You are a helpful poetry assistant.' },
      { role: 'user', content: 'Write a haiku about coding' }
    ]
  });
  console.log(response2.choices[0].message.content);
  
  console.log('\n3. With Specific Model:');
  console.log('-'.repeat(70));
  const response3 = await client.chat.completions.create({
    messages: [
      { role: 'user', content: 'Explain quantum computing in simple terms' }
    ],
    model: 'openai'
  });
  console.log(response3.choices[0].message.content);
  
  console.log('\n4. With Temperature Control:');
  console.log('-'.repeat(70));
  const response4 = await client.chat.completions.create({
    messages: [
      { role: 'user', content: 'Write a creative story opening' }
    ],
    temperature: 0.8,
    max_tokens: 100
  });
  console.log(response4.choices[0].message.content);
  
  console.log('\n5. Conversation Context:');
  console.log('-'.repeat(70));
  const response5 = await client.chat.completions.create({
    messages: [
      { role: 'system', content: 'You are a helpful assistant.' },
      { role: 'user', content: 'Tell me a joke.' },
      { role: 'assistant', content: 'Why do programmers prefer dark mode? Because light attracts bugs!' },
      { role: 'user', content: 'Tell me another one!' }
    ]
  });
  console.log(response5.choices[0].message.content);
  
  console.log('\n' + '='.repeat(70));
  console.log('Text generation examples completed!');
  console.log('='.repeat(70));
}

// Run the examples
main().catch(console.error);
